AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/SQLStatement+Select.swift

```diff
@@ -101,12 +72,81 @@ struct SelectStatement: SQLStatement {
             \(paginationInput.sqlStatement)
             """
         }
+        return SelectStatementMetadata(statement: sql,
+                                       columnMapping: columnMapping,
+                                       bindings: bindings)
+    }
+
+    /// Tuple that represents a pair of `columns` and `statements`
+    private typealias JoinStatements = ([String], [String], ColumnMapping)
+
+    /// Walk through the associations recursively to generate join statements.
+    ///
+    /// Implementation note: this should be revisited once we define support
+    /// for explicit `eager` vs `lazy` associations.
+    private static func joins(from schema: ModelSchema) -> JoinStatements {
+        var columns: [String] = []
+        var joinStatements: [String] = []
+        var columnMapping: ColumnMapping = [:]
+
+        func visitAssociations(node: ModelSchema, namespace: String = "root") {
+            for foreignKey in node.foreignKeys {
+                let associatedModelType = foreignKey.requiredAssociatedModel
+                let associatedSchema = associatedModelType.schema
+                let associatedTableName = associatedModelType.schema.name
+
+                // columns
+                let alias = namespace == "root" ? foreignKey.name : "\(namespace).\(foreignKey.name)"
+                let associatedColumn = associatedSchema.primaryKey.columnName(forNamespace: alias)
+                let foreignKeyName = foreignKey.columnName(forNamespace: namespace)
+
+                // append columns from relationships
+                columns += associatedSchema.columns.map { field -> String in
+                    columnMapping.updateValue((associatedSchema, field), forKey: "\(alias).\(field.name)")
+                    return field.columnName(forNamespace: alias) + " as " + field.columnAlias(forNamespace: alias)
+                }
+
+                let joinType = foreignKey.isRequired ? "inner" : "left outer"
+
+                joinStatements.append("""
+                \(joinType) join \(associatedTableName) as "\(alias)"
+                  on \(associatedColumn) = \(foreignKeyName)
+                """)
```

Comment:
Being ultra paranoid, do we care to filter out names of tables that might impact these statements.  Like... it would be bad if someone had a table called `Null` or `as`... maybe something to do for another PR.

Reply:
they are escaped with double quotes, so you can even pass a SQL statement there and nothing would happen.

that said, the names here are a result of the models, so only valid Swift identifiers can be passed here.